bot/common.py

```diff
@@ -452,6 +455,44 @@ def update_repos(project_path, git_config):
     return repos_dict
 
 
+def get_free_device():
+    tty = None
+    jlink = None
+    debuggers = subprocess.Popen('nrfjprog --com',
+                                 shell=True,
+                                 stdout=subprocess.PIPE
+                                 ).stdout.read().decode()
+
+    if sys.platform == "win32":
+        reg = "[0-9]+\s+COM[0-9]+(?=\s+.+)"
+    else:
+        reg = "[0-9]+\s+/dev/\w+(?=\s+.+)"
+
+    try:
+        debuggers = re.findall(reg, debuggers)
+        for d in debuggers:
+            srn = d.split()[0]
+            dev = d.split()[1]
+
+            if not devices_in_use.__contains__(srn):
+                devices_in_use.append(srn)
+                tty = dev
+                jlink = srn
+                break
+    except:
+        raise Exception("No jlinks found")
```

Comment:
I would prefer handling errors when they appear instead of catching everything with a try/except.